option.go

```diff
@@ -22,6 +22,18 @@ func Routes(rr []http.Route) OptionFunc {
 	}
 }
 
+// Middlewares option for adding generic middlewares to the default HTTP component.
+func Middlewares(mm []http.MiddlewareFunc) OptionFunc {
```

Comment:
i propose to turn this into a variadic function

Reply:
I followed the Routes func signature (above) and I wanted to make them both variadic but afraid to break compatibility :/ 
What do you think?

Reply:
I dont see how you break compatibility here?
i propose this signature on the new option:
```go
func Middlewares(mm ...http.MiddlewareFunc) OptionFunc {
```